ssz/src/main/java/tech/pegasys/artemis/util/backing/cache/ArrayCache.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.backing.cache;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.IntFunction;
+
+public final class ArrayCache<V> implements IntCache<V> {
```

Comment:
(Optional)
```suggestion
public final class ArrayIntCache<V> implements IntCache<V> {
```
or maybe:
```suggestion
public final class ArrayBackedIndexCache<V> implements IndexCache<V> {
```